### ISSUE TITLE
[BUG] The "TelegramException" (and children) can't be unpickled

### ISSUE THREAD

**Issue by Delgan** [python-telegram-bot/python-telegram-bot#2105]
### Steps to reproduce

Hi.  I encountered a small issue with an user because it does not seem possible to pickle `TelegramException` exceptions and their descendants. 

```python
import pickle
from telegram.error import TelegramException

error = TelegramException("Foobar")
pickled = pickle.dumps(error)      # OK
unpikcled = pickle.loads(pickled)  # NOK
# TypeError: __init__() missing 1 required positional argument: 'message'
```
This is actually not new, many Python users have encountered this kind of problem by implementing custom exceptions:
* [Cannot unpickle Exception subclass](https://stackoverflow.com/questions/41808912/cannot-unpickle-exception-subclass)
* [How to pickle inherited exceptions?](https://stackoverflow.com/questions/49715881/how-to-pickle-inherited-exceptions)
* [How to make a custom exception class with multiple init args pickleable](https://stackoverflow.com/questions/16244923/how-to-make-a-custom-exception-class-with-multiple-init-args-pickleable)

 
### Expected behaviour
The exception should be deserialized without issue.

### Actual behaviour
A `TypeError` is raised.

### Configuration
**Operating System:** Arch Linux
**Version of Python, python-telegram-bot & dependencies:**
```
python-telegram-bot 12.8 (0.5.3-0-gf31e971)
certifi 2020.06.20
Python 3.8.5 (default, Sep  5 2020, 10:50:12)  [GCC 10.2.0]
```

### Logs
Not relevant.

**Comment by Bibo-Joshi** [comment]
Hi. Thanks for reporting and for including the links to SO. However, I wonder what the use case of pickling exceptions would be. As far as I can see, TG errors should be handled at runtime and not be saved to file.

**Comment by Delgan** [comment]
In this case, the error occurred when trying to send the `TelegramException` to a `multiprocessing.Queue` (which requires that the objects can be (de)serialized using `pickle`). Actually, the `TelegramException` was not passed directly to the queue but rather contained in a `Record` object. Sending this record to a queue allowed parallel processing in a background process.

Surely the `exception` can be removed from the `Record` before sending it to the queue. Now that we figured out the issue, it's easy to find a solution. 

However, since the error occurred at the time of deserialization, it was difficult in our case to identify the source of the problem (since the background process cannot access the `Record`). I guess it may avoid similar incident in the future if exceptions can be safely pickled / unpickled.

**Comment by Bibo-Joshi** [comment]
Mh, okay … Well, even if the use case may be non-obvious, the fix by adding `__reduce__` should be small. Maybe you want to have a go at it yourself? It's hacktober after all 🙆‍♂️

**Comment by Delgan** [comment]
> Maybe you want to have a go at it yourself? It's hacktober after all ok_man

Sure. :smile: 
I can submit a PR during the afternoon.

>  Well, even if the use case may be non-obvious, the fix by adding `__reduce__` should be small.

The main downside I think is that it requires to implement `__reduce__` for each child exception using a custom `__init__()`. It does not suffice to implement it once for `TelegramException` unfortunately.

**Comment by Bibo-Joshi** [comment]
Closed by #2106
